packages/sdk-codegen/src/python.gen.spec.ts

```diff
@@ -467,14 +481,159 @@ class Workspace(model.Model):
     """
     can: Optional[MutableMapping[str, bool]] = None
     id: Optional[str] = None
-    projects: Optional[Sequence["Project"]] = None`)
+    projects: Optional[Sequence["Project"]] = None
+
+    def __init__(self, *,
+            can: Optional[MutableMapping[str, bool]] = None,
+            id: Optional[str] = None,
+            projects: Optional[Sequence["Project"]] = None):
+        self.can = can
+        self.id = id
+        self.projects = projects`)
+    })
+    it('re-orders required props to top', () => {
+      const type = apiTestModel.types.CreateDashboardFilter
+      const actual = gen.declareType(indent, type)
+      expect(actual).toEqual(`
+@attr.s(auto_attribs=True, init=False)
+class CreateDashboardFilter(model.Model):
+    """
+    Attributes:
+        dashboard_id: Id of Dashboard
+        name: Name of filter
+        title: Title of filter
+        type: Type of filter: one of date, number, string, or field
+        id: Unique Id
+        default_value: Default value of filter
+        model: Model of filter (required if type = field)
+        explore: Explore of filter (required if type = field)
+        dimension: Dimension of filter (required if type = field)
+        field: Field information
+        row: Display order of this filter relative to other filters
+        listens_to_filters: Array of listeners for faceted filters
+        allow_multiple_values: Whether the filter allows multiple filter values
+        required: Whether the filter requires a value to run the dashboard
+        ui_config: The visual configuration for this filter. Used to set up how the UI for this filter should appear.
+    """
+    dashboard_id: str
+    name: str
+    title: str
+    type: str
+    id: Optional[str] = None
```

Comment:
it was both `mypy` as well as the `attr` runtime that complained about actual class level "optional before required" ordering.

Reply:
Good to know